os/board/rtl8730e/src/component/os/tizenrt/ethernetif_tizenrt.c

```diff
@@ -174,10 +174,11 @@ err_t low_level_output(struct netdev *dev, uint8_t *data, uint16_t dlen)
 	if (sg_len) {
 #if CONFIG_WLAN
 #if defined(CONFIG_AS_INIC_AP)
-		/* Currently TizenRT only uses idx 0 on application layer, but KR4 splits STA and SOFTAP into idx 0 and idx 1, need to check which idx to send to */
+		/* If concurrent is disabled, TizenRT only uses idx 0 on application layer, but RTK driver splits STA and SOFTAP into idx 0 and idx 1, need to check which idx to send to */

```

Comment:
Hi @a-jhanwar , may I check if concurrent mode will be an option to configure in menuconfig in the future? In RTK's wifi layer, there are some code changes that need to be made if TizenRT is not using concurrent mode. If it is a config option, we will use a preprocessor to control the code flow during compilation.

Reply:
Hi @jwei5 ,
I am checking with team for menu config related query and update you.
In my opinion we should consider our implementation to handle following cases: 
1. There should be be some mechanism to disable this concurrent mode ( may be menu config or other) during compile time.
2. Even if this feature is on on compile time,
a) APP may decide to use this feature  or not
ex. During on boarding ( first time setup ) APP will use softap as its using currently.  There after APP may switch to concurrent mode .

What is your opinion on this? 
we can think for : do we need to provide new api or Add new parameter in existing APIs for RTK interface.

Regards,
Abhishek

Reply:
Hi @a-jhanwar ,
Actually, my concern is whether SoftAP will be fixed to wlan1 in the future, regardless of whether concurrent mode is enabled.
Before supporting concurrent mode, TizenRT only used wlan0 interface for both STA or SoftAP. However, STA and SoftAP interfaces are split into index 0 and index 1 respectively in our driver. This meant that we had to check which mode TizenRT was in, so that we could pass the packets to the correct index in our driver (thus the comment in line 177).
If the interfaces in TizenRT will be fixed to wlan0 for STA and wlan1 for SoftAP, then there will be no need for RTK to control the code flow, as we can check which index to use based on the interface name.